lib/rsolr/uri.rb

```diff
@@ -47,12 +38,37 @@ def params_to_solr(params, escape = true)
   # query strings faster.  Use this rather than the cgi.rb
   # version since it's faster.
   # (Stolen from Rack).
+  # 
+  # 2015-02
+  # The Rack stuff is from 
+  #  http://www.rubydoc.info/github/rack/rack/URI.encode_www_form_component
+  # We instead will rely on 
+  #  http://ruby-doc.org/stdlib-2.2.0/libdoc/uri/rdoc/URI.html#method-c-encode_www_form_component
+  #  which is from the Ruby stdlib.  Hence this method is deprecated.
+  # @deprecated
   def escape_query_value(s)
-    s.to_s.gsub(/([^ a-zA-Z0-9_.-]+)/u) {
-      '%'+$1.unpack('H2'*bytesize($1)).join('%').upcase
-    }.tr(' ', '+')
+    warn "[DEPRECATION] `escape_query_value` is deprecated.  Please use `URI.encode_www_form_component` instead."
+    URI.encode_www_form_component(s)
+#    s.to_s.gsub(/([^ a-zA-Z0-9_.-]+)/u) {
+#      '%'+$1.unpack('H2'*bytesize($1)).join('%').upcase
+#    }.tr(' ', '+')
   end
-  
+
+  # Return the bytesize of String; uses String#size under Ruby 1.8 and
+  # String#bytesize under 1.9.
+  # @deprecated  as bytesize was only used by escape_query_value which is itself deprecated
+  if ''.respond_to?(:bytesize)
```

Comment:
I wouldn't bother with asking for use cases. If it's a method on String already, they should always use that instead.